content/docs/rfcs/16/README.md

```diff
@@ -452,15 +452,14 @@ This method is part of the `store` API and the specific resources to retrieve ar
 
 #### Request
 
-```curl -d '{"jsonrpc":"2.0","id":"id","method":"get_waku_v2_store_v1_messages", "params":["", ["/waku/2/default-content/proto"]]}' --header "Content-Type: application/json" http://localhost:8545```
+```curl -d '{"jsonrpc":"2.0","id":"id","method":"get_waku_v2_store_v1_messages", "params":[["/waku/2/default-content/proto"]]}' --header "Content-Type: application/json" http://localhost:8545```
```

Comment:
If both are optional, how does it know which is pubsubTopic and which is contentFilters? Since they are both positional. Does it check type, so string implies first arg, and array second arg?

Reply:
Good question. I understood that you could only omit optional parameters towards the "end" of the list and otherwise have to explicitly pass `null` as an argument.

Reply:
Right, the optional parameters that are in the middle should be explicitly set to null, fixed it. Thanks!

Reply:
In general position arguments are pretty error-prone, especially once you try to make extensible/general APIs... might be time to consider `options map` where we instead specify set of key-value pairs for options? Means:

1) No need to think about positional arguments, less error-prone
2) Can specify arbitrary set of k-v params as arguments

In JSON RPC these are called named parameters, see https://www.jsonrpc.org/specification

Reply:
Interesting. To be clear, are you suggesting this as a workaround for https://github.com/status-im/nim-json-rpc/issues/90? 

Since `nim-json-rpc` does not support named parameters, we could achieve something similar by wrapping all arguments in another `args` object. This way the "outer" object will still remain unnamed and positional, but its composite fields could be named.

Reply:
Both approaches seem fine to me! 
If nim-json-rpc could support named-parameter passing then that would be a permanent solution.
In the meanwhile,  and till nim-json-rpc implements that feature, I can also follow @oskarth  suggested solution i.e., a Nim obj embodying all the parameters

Reply:
Oh right, eh, whatever is easiest then I suppose (current state?)

Reply:
Well, the easiest to use the current state, as it works fine, with positional arguments. I will also open an issue about what we have discussed in here, then depending on the priorities  it can be addressed now or later.

Reply:
If we decide to use Nim objects to handle named arguments, then this should be done for all the json-rpc APIs, right? @oskarth

Reply:
Here is the issue https://github.com/status-im/nim-waku/issues/533